### PR TITLE
Add a documentation page for compiler error CS7003

### LINKED ISSUE #1593: C# Compiler error: CS7003 is undocumented
# C# Compiler error: CS7003 is undocumented

# General

The compiler error: `error CS7003: Unexpected use of an unbound generic name`

### PR DESCRIPTION
**Pull request by sputier**
Fixes #1593

To this day, the compiler error CS7003 remains undocumented. This PR adds a documentation page for this error.

### CODE REVIEW

File: docs/csharp/language-reference/compiler-messages/cs7003.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# Compiler Error CS7003
+Unexpected use of an unbound generic name
+
+This error occurs if you use a generic type needing one parameter type without passing any generic parameter type name between the angle brackets. This use may be a variable declaration, or an object instanciation.
+
+## To correct this error  
+
```
Comment (rpetrusha) [inline docs/csharp/language-reference/compiler-messages/cs7003.md]: Since there aren't multiple items, this shouldn't be a list.

**Review by rpetrusha — CHANGES_REQUESTED** [review]
Thank you so much for documenting this compiler error, @sputier! I've left a number of suggestions for changes.

File: docs/csharp/language-reference/compiler-messages/cs7003.md (hunk 1 above)
Comment (sputier) [reply docs/csharp/language-reference/compiler-messages/cs7003.md]: Thanks for this remark @rpetrusha. I just pushed a commit to fix this.

**Comment by mairaw** [comment]
This page should also be added to the TOC file: https://github.com/dotnet/docs/blob/master/docs/csharp/language-reference/compiler-messages/toc.md. Let us know if you'd like to take care of that for you.

**Comment by sputier** [comment]
Thanks for the remark @mairaw . I just pushed an update to the PR.

**Review by rpetrusha — APPROVED** [review]

**Comment by rpetrusha** [comment]
Thanks for updating the TOC, @sputier. This looks good. We'll merge your PR now. The topic should appear on docs.microsoft.com in the next day or two.

**Pull request merged**